libnxz.h

```diff
@@ -0,0 +1,147 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <stdint.h>
+#include <stddef.h>
+
+#ifndef _LIBNXZ_H
+#define _LIBNXZ_H
+
+#ifndef __unused
+#  define __unused __attribute__((unused))
+#endif
+
+/* zlib crc32.c, crc32_ppc.c and adler32.c */
+extern ulong nx_adler32(ulong adler, const char *buf, uint len);
+extern ulong nx_adler32_z(ulong adler, const char *buf, size_t len);
+extern ulong nx_adler32_combine(ulong adler1, ulong adler2, uint64_t len2);
+extern ulong nx_crc32(ulong crc, const unsigned char *buf, uint64_t len);
+extern ulong nx_crc32_combine(ulong crc1, ulong crc2, uint64_t len2);
+extern ulong nx_crc32_combine64(ulong crc1, ulong crc2, uint64_t len2);
+extern ulong crc32_ppc(ulong crc, unsigned char *buffer, unsigned len);
+
+/* nx_deflate.c */
+extern int nx_deflateInit_(void *strm, int level, const char *version,
+			   int stream_size);
+extern int nx_deflateInit2_(void *strm, int level, int method, int windowBits,
+			    int memLevel __unused, int strategy,
+			    const char *version __unused,
+			    int stream_size __unused);
+#define nx_deflateInit(strm, level) nx_deflateInit_((strm), (level), \
+					ZLIB_VERSION, (int)sizeof(z_stream))
+extern int nx_deflate(void *strm, int flush);
+extern int nx_deflateEnd(void *strm);
+extern ulong nx_deflateBound(void *strm, ulong sourceLen);
+extern int nx_deflateSetHeader(void *strm, void *head);
+extern int nx_deflateCopy(void *dest, void *source);
+extern int nx_deflateReset(void *strm);
+extern int nx_deflateResetKeep(void *strm);
+extern int nx_deflateSetDictionary(void *strm, const unsigned char *dictionary,
+				   uint dictLength);
+
+/* nx_inflate.c */
+extern int nx_inflateInit_(void *strm, const char *version, int stream_size);
+extern int nx_inflateInit2_(void *strm, int windowBits, const char *version,
+			    int stream_size);
+#define nx_inflateInit(strm) nx_inflateInit_((strm), ZLIB_VERSION, \
+					(int)sizeof(z_stream))
+extern int nx_inflate(void *strm, int flush);
+extern int nx_inflateEnd(void *strm);
+extern int nx_inflateCopy(void *dest, void *source);
+extern int nx_inflateGetHeader(void *strm, void *head);
+extern int nx_inflateResetKeep(void *strm);
+extern int nx_inflateSetDictionary(void *strm, const unsigned char *dictionary,
+				   uint dictLength);
+extern int nx_inflateReset(void *strm);
+
+/* nx_compress.c */
+extern int nx_compress2(unsigned char *dest, ulong *destLen,
+			const unsigned char *source, ulong sourceLen,
+			int level);
+extern int nx_compress(unsigned char *dest, ulong *destLen,
+		       const unsigned char *source, ulong sourceLen);
+extern ulong nx_compressBound(ulong sourceLen);
+
+/* nx_uncompr.c */
+extern int nx_uncompress2(unsigned char *dest, ulong *destLen,
+			  const unsigned char *source, ulong *sourceLen);
+extern int nx_uncompress(unsigned char *dest, ulong *destLen,
+			 const unsigned char *source, ulong sourceLen);
+
+
+#ifdef ZLIB_API
```

Comment:
This is exposing an internal macro to the user.
Is this really intentional?
Can't we export these functions without depending on this macro?

Reply:
Was intentional because we don't need these functions if the macro is not defined, but yes we can export them without it, I'll change that.